### ISSUE TITLE
Gateway index pages link to wrong places when using CID-as-subdomain dweb.link

### ISSUE THREAD

**Issue by fiatjaf** [ipfs/go-ipfs#5309]
Steps to reproduce:

```shell
fiatjaf@menger ~> mkdir -p a/b/c
fiatjaf@menger ~> ipfs add --pin=false -r --cid-version=1 a
added zdj7WbTaiJT1fgatdet9Ei9iDB5hdCxkbVyhyh8YTUnXMiwYi a/b/c
added zdj7WeXRaRhcsffmWMAYkfa6x74v3L3VkvGE7EdoGaGvz8Dkq a/b
added zdj7WXZ8Z4tQu1yVqECwvSUEQZTJsSCw1y38MzbXpuwYf3eJT a
 0 B / 12.00 KiB [------------------------------------------------------------]   0.00
# get base32 cid on http://cid-utils.ipfs.team/#zdj7WXZ8Z4tQu1yVqECwvSUEQZTJsSCw1y38MzbXpuwYf3eJT
fiatjaf@menger ~> curl -s bafybeia7sgwof4y6sbaqcqvhfsysegwigqtksbm6irbzgnw4luff5htlxa.ipfs.dweb.link | grep 'href='                                                 
            <a href="/ipfs/bafybeia7sgwof4y6sbaqcqvhfsysegwigqtksbm6irbzgnw4luff5htlxa/">..</a>
            <a href="/ipfs/bafybeia7sgwof4y6sbaqcqvhfsysegwigqtksbm6irbzgnw4luff5htlxa/b">b</a>
```

As you can see, the anchors on that page, http://bafybeia7sgwof4y6sbaqcqvhfsysegwigqtksbm6irbzgnw4luff5htlxa.ipfs.dweb.link/, are linking to non-existing paths, such as http://bafybeia7sgwof4y6sbaqcqvhfsysegwigqtksbm6irbzgnw4luff5htlxa.ipfs.dweb.link/ipfs/bafybeia7sgwof4y6sbaqcqvhfsysegwigqtksbm6irbzgnw4luff5htlxa/b, when they should link just to `/b`.

**Comment by Stebalien** [comment]
So, the CID-as-subdomain dweb.link feature is implemented using a separate nginx reverse proxy. IPFS has no idea what's going on here.

I believe the simplest fix would be to use relative links in generated directory indexes.

**Comment by lidel** [comment]
cc https://github.com/ipfs/ipfs/issues/337

**Comment by Stebalien** [comment]
Fixed in #6498.
